File: codes/week16/진현/26.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int prev_num = nums[0], ret = 1;
        for (int i=1; i<nums.size(); i++) {
            int num = nums[i];

            if (prev_num == num) {
                nums[i] = INT_MAX;
            }
            else {
                prev_num = nums[i];
                ret++;
            }
        }

        sort(nums.begin(), nums.end());
        return ret;
    }
};

File: codes/week16/지원/26.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        curr_idx = -1
        curr_num = None
        n = len(nums)

        for num in nums:
            if num != curr_num:
                curr_num = num
                curr_idx += 1
                nums[curr_idx] = curr_num

        return curr_idx + 1